### ISSUE TITLE
SRFI 64 testing bugs on R6RS mode

### ISSUE THREAD

**Issue by ktakashi** [larcenists/larceny#725]
This raises `&syntax`

```scheme
(import (rnrs) (srfi :64))

(test-begin "foo")

(test-error "foo" condition? (car 'a))

(test-end)
#|
Syntax violation: %test-error

Invalid form

Form: (%test-error condition? (car 'a))

(trace)
|#
```

And the this goes into infinite loop.

```scheme
(import (rnrs) (srfi :64))

(test-begin "foo")

(test-assert (car 'a))

(test-end)

#|
Error: no handler for exception #<record &compound-condition>

Error: no handler for exception #<record &compound-condition>

(omit)
|#
```

Environment:
Larceny v0.98 "General Ripper" (Mar  7 2015 01:06:26, precise:Linux:unified)

**Comment by WillClinger** [comment]
Thank you for your bug report.

The `&syntax` exception is a bug in the reference implementation for SRFI 64.  In implementations other than Kawa and MzScheme, the `test-error` macro expands into a use of the `%test-error` macro with only two macro arguments; it requires three.  It looks as though the easiest fix is to add `larceny` to the `cond-expand` clause that special-cases Kawa and MzScheme, but that wouldn't fix the bug for other systems.

I was unable to duplicate the infinite loop on my Linux machine.

For your second example, I get a `"car: not a pair: a"` error message, which seems like the result to be expected in systems other than Kawa and MzScheme, which are special-cased and might handle exceptions like this.  Once again, it looks as though the easiest fix is to add `larceny` to the list of systems that get the special treatment, but that wouldn't fix the bug for other systems.

**Comment by WillClinger** [comment]
My mistake: I can duplicate the infinite loop in `--r6rs` mode, which should have been obvious to me from the title of the bug report.

Fixed by changeset 455f54ec290a56ab0cb3cad66a2fbc64c2d4cd62
